docs/book/config.md

```diff
@@ -50,55 +53,84 @@ included as part of your application environment.
+        //         'graphApiVersion' => 'v2.10',
+        //     ],
         // ],
 
         // GitHub
         // 'github' => [
-        //     'clientId' => '{github-client-id}',
-        //     'clientSecret' => '{github-client-secret}',
-        //     'redirectUri' => '', // based on the auth_path + production route; must be fully qualifed
+        //     'provider' => Provider\Github::class,
+        //     'options' => [
+        //         'clientId' => '{github-client-id}',
+        //         'clientSecret' => '{github-client-secret}',
+        //         'redirectUri' => '', // based on the auth_path + production route; must be fully qualifed
+        //     ],
         // ],
 
         // Google
         // 'google' => [
-        //     'clientId' => '{google-client-id}',
-        //     'clientSecret' => '{google-client-secret}',
-        //     'redirectUri' => '', // based on the auth_path + production route; must be fully qualifed
-        //     'hostedDomain' => '', // scheme + domain of your app
+        //     'provider' => Provider\Google::class,
+        //     'options' => [
+        //         'clientId' => '{google-client-id}',
+        //         'clientSecret' => '{google-client-secret}',
+        //         'redirectUri' => '', // based on the auth_path + production route; must be fully qualifed
+        //         'hostedDomain' => '', // scheme + domain of your app
+        //     ],
         // ],
 
         // Instagram
         // 'instagram' => [
-        //     'clientId' => '{instagram-client-id}',
-        //     'clientSecret' => '{instagram-client-secret}',
-        //     'redirectUri' => '', // based on the auth_path + production route; must be fully qualifed
-        //     'host' => 'https://api.instagram.com', // Optional; this is the default
+        //     'provider' => Provider\Instagram::class,
+        //     'options' => [
+        //        'clientId' => '{instagram-client-id}',
+        //        'clientSecret' => '{instagram-client-secret}',
+        //        'redirectUri' => '', // based on the auth_path + production route; must be fully qualifed
+        //        'host' => 'https://api.instagram.com', // Optional; this is the default
+        //     ],
         // ],
 
         // LinkedIn
         // 'linkedin' => [
-        //     'clientId' => '{linkedin-client-id}',
-        //     'clientSecret' => '{linkedin-client-secret}',
-        //     'redirectUri' => '', // based on the auth_path + production route; must be fully qualifed
+        //     'provider' => Provider\LinkedIn::class,
+        //     'options' => [
+        //         'clientId' => '{linkedin-client-id}',
+        //         'clientSecret' => '{linkedin-client-secret}',
+        //         'redirectUri' => '', // based on the auth_path + production route; must be fully qualifed
+        //     ],
+        // ],
+
+        // Customized
+        // 'custom' => [
```

Comment:
I'd add an explanation below this of the "custom" provider here - that the purpose is to allow configuring any provider that extend the League `AbstractProvider`, including custom implementations or extensions; this will make it more clear to end users that we support the full-spectrum of the League OAuth2 client library.

Reply:
```suggestion
        // It's possible to configure any provider that extend the League AbstractProvider, including custom 
        // implementations or extensions. This example uses GenericProvider which can be used with any OAuth 
        // 2.0 Server that uses Bearer tokens.
        // 'custom' => [
```

Reply:
Good!